backend/dataall/core/stacks/services/stack_service.py

```diff
@@ -1,25 +1,55 @@
 import os
 
 import requests
+import logging
 
-from dataall.core.environment.db.environment_repositories import EnvironmentRepository
+from dataall.core.permissions.services import environment_permissions
+from dataall.core.permissions.services.resource_policy_service import ResourcePolicyService
+from dataall.core.stacks.aws.cloudformation import CloudFormation
+from dataall.core.stacks.services.keyvaluetag_service import KeyValueTagService
 from dataall.core.tasks.service_handlers import Worker
 from dataall.base.config import config
 from dataall.base.context import get_context
-from dataall.core.environment.db.environment_models import Environment
 from dataall.core.stacks.aws.ecs import Ecs
 from dataall.core.stacks.db.stack_repositories import StackRepository
 from dataall.core.stacks.db.stack_models import Stack
 from dataall.core.tasks.db.task_models import Task
 from dataall.base.utils import Parameter
+from dataall.core.stacks.aws.cloudwatch import CloudWatch
+from dataall.base.db.exceptions import AWSResourceNotFound
+from dataall.base.db.exceptions import RequiredParameter
+from dataall.core.stacks.db.target_type_repositories import TargetType
+
+
+log = logging.getLogger(__name__)
+
+
+class StackRequestVerifier:
+    @staticmethod
+    def verify_get_and_describe_params(env_uri, stack_uri):
+        if not env_uri:
+            raise RequiredParameter('Environment URI is required')
+        if not stack_uri:
+            raise RequiredParameter('Stack URI is required')
+
+    @staticmethod
+    def validate_update_tag_input(data):
+        if not data.get('targetUri'):
+            raise ValueError('targetUri is required')
+
+    @staticmethod
+    def verify_target_type_and_uri(target_type, target_uri):
+        if not target_uri:
+            raise RequiredParameter('targetUri')
+        if not target_type:
+            raise RequiredParameter('targetType')
 
 
 class StackService:
     @staticmethod
-    def get_stack_with_cfn_resources(targetUri: str, environmentUri: str):
+    def get_stack_with_cfn_resources(targetUri: str, env):
```

Comment:
This is a pretty important function, very minor, can we add type hints for env:Environment?

Reply:
Done: [5ddf7b3](https://github.com/data-dot-all/dataall/pull/1181/commits/5ddf7b3340e1540b445a53a8052de37291b22483)